src/core/pb_hkdf_sha256.rs

```diff
@@ -17,6 +17,21 @@ use k256::sha2::Sha256;
 #[display("{}", self.description())]
 pub struct PbHkdfSha256 {}
 
+impl PbHkdfSha256 {
+    pub fn hkdf_key_agreement(
```

Comment:
the function `hkdf_key_agreement` should probably be renamed `derive_key` or `expand` it is a static func on `PbHkdfSha256` and it has nothing to do with key agreement, the function itself just performs key derivation by expansion.